pySDC/playgrounds/HotRod/accuracy_check.py

```diff
@@ -0,0 +1,188 @@
+    controller = controller_nonMPI(num_procs=1, controller_params=controller_params, description=description)
+
+    # get initial values on finest level
+    P = controller.MS[0].levels[0].prob
+    uinit = P.u_exact(t0)
+
+    # call main function to get things done...
+    uend, stats = controller.run(u0=uinit, t0=t0, Tend=Tend)
+
+    results = {
+        'e_embedded': sort_stats(filter_stats(stats, type='e_embedded'), sortby='time')[-1][1],
+        'e_extrapolated': sort_stats(filter_stats(stats, type='e_extrapolated'), sortby='time')[-1][1],
+        var: val,
+    }
+
+    return results
+
+
+def mulitple_runs(ax, k=5):
+    """
+    A simple test program to compute the order of accuracy in time
+    """
+
+    # assemble list of dt
+    dt_list = 0.05 * 10.**-(np.arange(20) / 10.)
+
+    # perform first test
+    res = single_run(var='dt', val=dt_list[0], k=k)
+    for key in res.keys():
+        res[key] = [res[key]]
+
+    # perform rest of the tests
+    for i in range(1, len(dt_list)):
+        res_ = single_run(var='dt', val=dt_list[i], k=k)
+        for key in res_.keys():
+            res[key].append(res_[key])
+
+    # visualize results
+    plot(res, ax, k)
+
+
+def plot(res, ax, k):
+    keys = ['e_embedded', 'e_extrapolated']
+    ls = ['-', '--']
+    color = plt.rcParams['axes.prop_cycle'].by_key()['color'][k - 3]
+
+    for i in range(len(keys)):
+        order = get_accuracy_order(res, key=keys[i], order=k)
+        if i == 0:
+            label = rf'$k={{{np.mean(order):.2f}}}$'
+        else:
+            label = None
+        ax.loglog(res['dt'], res[keys[i]], color=color, ls=ls[i], label=label)
+
+    ax.set_xlabel(r'$\Delta t$')
+    ax.set_ylabel(r'$\epsilon$')
+    ax.legend(frameon=False)
+
+
+def get_accuracy_order(results, key='e_embedded', order=5):
+    """
+    Routine to compute the order of accuracy in time
+
+    Args:
+        results: the dictionary containing the errors
+
+    Returns:
+        the list of orders
+    """
+
+    # retrieve the list of dt from results
+    assert 'dt' in results, 'ERROR: expecting the list of dt in the results dictionary'
+    dt_list = sorted(results['dt'], reverse=True)
+
+    order = []
+    # loop over two consecutive errors/dt pairs
+    for i in range(1, len(dt_list)):
+        # compute order as log(prev_error/this_error)/log(this_dt/old_dt) <-- depends on the sorting of the list!
+        tmp = np.log(results[key][i] / results[key][i - 1]) / np.log(dt_list[i] / dt_list[i - 1])
+        order.append(tmp)
+
+    return order
+
+
+def main():
+    ks = [5, 4, 3, 2]
+    fig, ax = plt.subplots(1, 1)
+    for i in range(len(ks)):
+        k = ks[i]
+        mulitple_runs(k=k, ax=ax)
+    ax.plot([None, None], color='black', label=r'$\epsilon_\mathrm{embedded}$', ls='-')
+    ax.plot([None, None], color='black', label=r'$\epsilon_\mathrm{extrapolated}$', ls='--')
+    ax.legend(frameon=False)
+    fig.savefig('error_order.pdf', bbox_inches='tight')
+
```

Comment:
You could start adding `assert`-statements to the different functions to help you testing the results.